sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/parquet/ParquetTable.java

```diff
@@ -122,4 +127,27 @@ public IsBounded isBounded() {
   public ProjectSupport supportsProjects() {
     return ProjectSupport.WITH_FIELD_REORDERING;
   }
+
+  private String resolveFilePattern(String location) {
+    try {
+      MatchResult match = FileSystems.match(location);
+      if (match.status() == MatchResult.Status.OK && !match.metadata().isEmpty()) {
+        MatchResult.Metadata metadata = match.metadata().get(0);
+        if (metadata.resourceId().isDirectory()) {
+          String dirPath = metadata.resourceId().toString();
+          if (dirPath.endsWith("/")) {
+            return dirPath + "*";
+          } else {
+            return dirPath + "/*";
+          }
+        }
+      }
+    } catch (IOException e) {
+      LOG.warn(
```

Comment:
buildIOReader happens at pipeline expansion time, and it is totally valid use case that submission VM not having access to the filesystem location (consider user submit pipeline locally and run in Dataflow).

How about not rely on FileSystems.match call to resolveFilePattern? For example, if location has '*', consider it as a glob; if location ends with pqt, or .parquet, consider it as a single file

Reply:
Thank you @Abacn This is good call out that i  forgot it. Let me change this function behavior in a way If location ends with "/", it's treated as a directory. Otherwise, it's treated as a glob or a single file path.

```java
  private String resolveFilePattern(String location) {
    if (location.endsWith("/")) {
      return location + "*";
    }
    return location;
  }
```